app/main.py

```
@@ -1,3 +1,56 @@
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: float) -> None:
+        self.km = km
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __add__(self, other : "Distance") -> "Distance":
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        return Distance(self.km + other)
+
+    def __iadd__(self, other : "Distance") -> "Distance":
+        if isinstance(other, Distance):
+            self.km += other.km
+        else:
+            self.km += float(other)
+        return self
+
+    def __mul__(self, other : "Distance") -> "Distance":
+        if isinstance(other, Distance):
+            return None
```

Returning `None` for unsupported multiplication between two `Distance` objects is not ideal. According to best practices and likely checklist requirements, you should raise a `TypeError` instead of returning `None` for unsupported operations.